### ISSUE TITLE
expo-secure-store broken on M1 / Apple Silicon Macs

### ISSUE THREAD

**Issue by LinusU** [expo/expo#21153]
### Summary

Both `getItemAsync` and `setItemAsync` fails with the following error when running an iOS app on M1/Apple Silicon Macs. We have been able to reproduce this in both the a dev client build, and an app store build, on two different MacBooks (M1 Air, M2 Air).

```text
Error: The operation couldn’t be completed. (OSStatus error -34018.)
```

~~This started happening when we upgraded to SDK 46. Upgrading to SDK 47 did not resolve the issue.~~

This happens on macOS 13.1, running the same app on macOS 12.1 works.

### What platform(s) does this occur on?

iOS

### Environment

```text
  expo-env-info 1.0.2 environment info:
    System:
      OS: macOS 13.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
      Yarn: 1.22.19 - ~/coding/loco/order/node_modules/.bin/yarn
      npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
      Watchman: 2022.12.26.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^47.0.0 => 47.0.13 
      react: 18.1.0 => 18.1.0 
      react-dom: 18.1.0 => 18.1.0 
      react-native: 0.70.5 => 0.70.5 
    npmGlobalPackages:
      expo-cli: 6.0.1
    Expo Workflow: managed
```

### Minimal reproducible example

Create a new app:

```sh
npx create-expo-app my-app
cd my-app
npx expo-cli add expo-secure-store
```

Replace `App.js`:

```js
import { useEffect, useState } from 'react'
import { setItemAsync } from 'expo-secure-store'
import { Text, View } from 'react-native'

export default function App() {
  const [error, setError] = useState(null)

  useEffect(() => {
    setItemAsync('test', 'test').catch(setError)
  }, [])

  return (
    <View style={{ padding: 64 }}>
      <Text>Error: {String(error)}</Text>
    </View>
  )
}
```

Configure EAS build:

```sh
npx eas-cli build:configure
```

(select iOS, and then accept all defaults)

Create a build:

```sh
npx eas-cli build --profile=development
```

(select iOS, and then accept all defaults)

Start the dev server:

```
expo start --dev-client
```

Download the IPA from Expo, open it on your Apple Silicon MacBook (right click and select open, then click open again).

Select your dev-server, then observe the error :boom:

**Comment by brentvatne** [comment]
hey @LinusU - is this only happening on simulator for you? not on a device? if so, it seems likely that there is a bug in the simulator on arm64

**Comment by LinusU** [comment]
@brentvatne it's not using the simulator, Apple Silicon Macs have native support for running iOS apps from the App Store and  your own .ipa files.

Here is some more info: https://developer.apple.com/documentation/apple-silicon/running-your-ios-apps-in-macos

**Comment by LinusU** [comment]
The diff inside `packages/expo-secure-store` for `sdk-45..sdk-46` looks empty, so my guess is that it's something in the base SDK that has changed 🤔

**Comment by LinusU** [comment]
I've updated the OP with some new information, this doesn't seem to be related to SDK version, instead it started appearing with a newer version of macOS.

It works with macOS 12.1, it doesn't work with macOS 13.1.

**Comment by eliasnorstrom** [comment]
Got on the same issue on macOS 13.1.
Did upgrade to macOS 13.3 (22E252) and all of the apps works again! 🎉 

This issue should be related to macOS 13.1 but it should be resolved in at least 13.3, but even maybe in 13.2.
My spec:
Modelname:	MacBook Air
Modelidentifier:	Mac14,2
Modelnumber:	Z15S000EEKS/A
Processor:	Apple M2

**Comment by LinusU** [comment]
Seems to be working in macOS 13.2.1 (22D68) as well 🎉 

I guess we can close this out as a macOS bug then? 😅
